bin/crunchy-exporter/start.sh

```diff
@@ -0,0 +1,193 @@
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source /opt/cpm/bin/common_lib.sh
+enable_debugging
+
+export PG_EXP_HOME=$(find /opt/cpm/bin/ -type d -name 'postgres_exporter*')
+export PG_DIR=$(find /usr/ -type d -name 'pgsql-*')
+POSTGRES_EXPORTER_PIDFILE=/tmp/postgres_exporter.pid
+CONFIG_DIR='/opt/cpm/conf'
+QUERIES=(
+    queries_backrest
+    queries_common
+    queries_per_db
+)
+
+function trap_sigterm() {
+    echo_info "Doing trap logic.."
+
+    echo_warn "Clean shutdown of postgres-exporter.."
+    kill -SIGINT $(head -1 ${POSTGRES_EXPORTER_PIDFILE?})
+}
+
+# Set default env vars for the postgres exporter container
+set_default_postgres_exporter_env() {
+    if [[ ! -v POSTGRES_EXPORTER_PORT ]]
+    then
+        export POSTGRES_EXPORTER_PORT="9187"
+        default_exporter_env_vars+=("POSTGRES_EXPORTER_PORT=${POSTGRES_EXPORTER_PORT}")
+    fi
+}
+
+# Set default PG env vars for the exporter container
+set_default_pg_exporter_env() {
+
+    if [[ ! -v EXPORTER_PG_HOST ]]
+    then
+        export EXPORTER_PG_HOST="127.0.0.1"
+        default_exporter_env_vars+=("EXPORTER_PG_HOST=${EXPORTER_PG_HOST}")
+    fi
+
+    if [[ ! -v EXPORTER_PG_PORT ]]
+    then
+        export EXPORTER_PG_PORT="5432"
+        default_exporter_env_vars+=("EXPORTER_PG_PORT=${EXPORTER_PG_PORT}")
+    fi
+
+    if [[ ! -v EXPORTER_PG_DATABASE ]]
+    then
+        export EXPORTER_PG_DATABASE="postgres"
+        default_exporter_env_vars+=("EXPORTER_PG_DATABASE=${EXPORTER_PG_DATABASE}")
+    fi
+
+    if [[ ! -v EXPORTER_PG_USER ]]
+    then
+        export EXPORTER_PG_USER="ccp_monitoring"
+        default_exporter_env_vars+=("EXPORTER_PG_USER=${EXPORTER_PG_USER}")
+    fi
+
+    env_check_err "EXPORTER_PG_PASSWORD"
+}
+
+trap 'trap_sigterm' SIGINT SIGTERM
+
+set_default_postgres_exporter_env
+
+if [[ ! -v DATA_SOURCE_NAME ]]
+then
+    set_exporter_pg_credentials
+    set_default_pg_exporter_env
+    if [[ ! -z "${EXPORTER_PG_PARAMS}" ]]
+    then
+        EXPORTER_PG_PARAMS="?${EXPORTER_PG_PARAMS}"
+    fi
+    export DATA_SOURCE_NAME="postgresql://${EXPORTER_PG_USER}:${EXPORTER_PG_PASSWORD}\
+@${EXPORTER_PG_HOST}:${EXPORTER_PG_PORT}/${EXPORTER_PG_DATABASE}${EXPORTER_PG_PARAMS}"
+fi
+
+
+
+if [[ ! ${#default_exporter_env_vars[@]} -eq 0 ]]
+then
+    echo_info "Defaults have been set for the following exporter env vars:"
+    echo_info "[${default_exporter_env_vars[*]}]"
+fi
+
+# Check that postgres is accepting connections.
+echo_info "Waiting for PostgreSQL to be ready.."
+while true; do
+    ${PG_DIR?}/bin/pg_isready -d ${DATA_SOURCE_NAME}
+    if [ $? -eq 0 ]; then
+        break
+    fi
+    sleep 2
+done
```

Comment:
I see this here, and our `pgisready` func defined in the common file. I think we should either use this or that. I'm fine with this one.

Reply:
Works for me. removed `pgisready()` from the common_lib.sh